### ISSUE TITLE
Extremely high losses

### ISSUE THREAD

**Issue by DenysLi** [keras-team/keras#7769]
_solved_

Hi,

I am making depth recognition, and suddenly got a problem with high losses (here is mean percentage). They are approaching to NaN in a few epochs.
![123](https://user-images.githubusercontent.com/23259843/29838116-e2365be0-8d02-11e7-8953-219a8faaa2b0.PNG)
Data is fitted by custom generator (it works correctly)
```
def generator(batch_size=32):
    while True:
        batch_input, batch_output = postProcess()
        yield batch_input, batch_output   
```
Where postProcess() returns:
Inputs are [Batch Size, Width*2, Height,    3-Channels]
Outputs a [Batch Size, Width/2, Height/2, 1-Channel]

Inputs are merged through Width-axis.
Example of input-output

![00002](https://user-images.githubusercontent.com/23259843/29837949-5f7cd684-8d02-11e7-923b-d36408d9ccff.png)

![00002](https://user-images.githubusercontent.com/23259843/29837761-c7421ece-8d01-11e7-9e4d-40ad96f2f67b.png)

I have tried all optimizers, decreasing learning rate up to 1e-100, changing batch size, number of layers and their depth..

 ```
   model = Sequential()
    # # # First Layer # # #
    model.add(Convolution2D(num_filters, (5,5),
                            padding='valid',
                            strides=2,
                            data_format='channels_last',
                            input_shape=(cut_size*2, cut_size,3) ) )
    model.add(ZeroPadding2D(padding=(1,1), data_format=None))
    model.add(BatchNormalization())
    model.add(Activation('relu'))
    model.add(MaxPooling2D((2,1)))
    # # # Second Layer    # # #
    model.add(Convolution2D(num_filters, (3,3)))
    model.add(ZeroPadding2D(padding=(1,1), data_format=None))
    model.add(BatchNormalization())
    model.add(Activation('relu'))
    # # # Fully Connected # # #
    model.add(Dense(1))
    model.compile(loss='mean_absolute_percentage_error', optimizer=SGD(lr=0.01, momentum=0.0, decay=0.0, nesterov=False))
```
Also i checked min and max and they look like truth: (0..255) for output and ~ (-1.4.. 1.4 ) for input. I dont know how this can lead to such loss.

Any hints how to solve this? Maybe you encountered same before?
Thanks in advance,
Dennis

remarque: I solved this, the problem is not in Keras. It is scy-py misc library which saves and than reads incorrectly non-integer arrays as images. I changed np.array(A, dtype=np float) to **np.array(A, dtype=np.uint8)**

**Comment by je-santos** [comment]
Same problem here
